src/main/webapp/app/shared/modal/ModifyCancerTypeModal.tsx

```diff
@@ -0,0 +1,180 @@
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+import { SimpleConfirmModal } from './SimpleConfirmModal';
+import CancerTypeSelect, { CancerTypeSelectOption } from '../select/CancerTypeSelect';
+import { Col, Container, Row } from 'reactstrap';
+import { CancerType } from '../model/firebase/firebase.model';
+import { getCancerTypeName } from '../util/utils';
+import { IRootStore } from 'app/stores';
+import { componentInject } from '../util/typed-inject';
+import { ICancerType } from '../model/cancer-type.model';
+import { FaExclamationCircle } from 'react-icons/fa';
+
+export interface IModifyCancerTypeModalProps extends StoreProps {
+  includedCancerTypes: CancerType[];
+  excludedCancerTypes: CancerType[];
+  onConfirm: (includedCancerTypes: CancerType[], excludedCancerTypes: CancerType[]) => void;
+  onCancel: () => void;
+  open?: boolean;
+}
+
+function ModifyCancerTypeModal({
+  includedCancerTypes,
+  excludedCancerTypes,
+  onConfirm,
+  onCancel,
+  open = false,
+  searchCancerTypes,
+}: IModifyCancerTypeModalProps) {
+  return open ? (
+    <ModifyCancerTypeModalContent
+      includedCancerTypes={includedCancerTypes}
+      excludedCancerTypes={excludedCancerTypes}
+      onConfirm={onConfirm}
+      onCancel={onCancel}
+      searchCancerTypes={searchCancerTypes}
+    />
+  ) : (
+    <></>
+  );
+}
+
+function ModifyCancerTypeModalContent({
+  includedCancerTypes,
+  excludedCancerTypes,
+  onConfirm,
+  onCancel,
+  searchCancerTypes,
+}: Omit<IModifyCancerTypeModalProps, 'open'>) {
+  const [included, setIncluded] = useState<CancerTypeSelectOption[]>([]);
+  const [excluded, setExcluded] = useState<CancerTypeSelectOption[]>([]);
+
+  function getCancerTypeFromCancerTypeSelectOption(cancerTypeSelectOption: CancerTypeSelectOption) {
+    const cancerType = new CancerType();
+    cancerType.code = cancerTypeSelectOption.code;
+    cancerType.mainType = cancerTypeSelectOption.mainType;
+    cancerType.subtype = cancerTypeSelectOption.subtype;
+    return cancerType;
+  }
+
+  async function getICancerTypeFromCancerType(cancerType: CancerType) {
+    try {
+      const iCancerType = await searchCancerTypes({ query: cancerType.subtype ? cancerType.subtype : cancerType.mainType });
+      return iCancerType['data'][0];
+    } catch {
+      return null;
```

Comment:
I think this will be wrong if we return null. I would rather disable the functionality of modifying when error happens.

Reply:
I disabled the dropdowns and the confirm button, added an error message, and a retry button to  retry fetching the data